cpp/calculate-demand.cpp

```diff
@@ -73,45 +59,79 @@ long readUnsatisfiedDemand(std::string targetFilename) {
   return atol(metadataValue);
 }
 
-long calculateUnsatisfiedDemand(std::string targetFilename, std::string demoFilename, std::vector<std::string> facilities, std::vector<float> capacities) {
-
-  // generate target dataset based on demoDataset
-  GDALDataset* demoDataset = openRaster(demoFilename);
-  GDALRasterBand* demoBand = demoDataset->GetRasterBand(1);
-  assert(demoBand->GetRasterDataType() == GDT_Float32);
-  int xBlockSize, yBlockSize;
-  demoBand->GetBlockSize(&xBlockSize, &yBlockSize);
-  GDALDataset* targetDataset = copyRaster(demoDataset, targetFilename, xBlockSize, yBlockSize);
-  GDALRasterBand* targetBand = targetDataset->GetRasterBand(1);
-  closeRaster(demoDataset);
+long calculateUnsatisfiedDemand(std::string targetFilename, std::string demoFilename, float saturation, std::vector<std::string> facilities, std::vector<float> capacities) {
+
+  #ifdef BENCHMARK
```

Comment:
Preprocessor directives are usually indented independently and not with the code https://google.github.io/styleguide/cppguide.html#Preprocessor_Directives